CONTRIBUTING.md

```diff
@@ -47,100 +45,117 @@ Also, remove the "wrench" icon (🔧) from the TOC and the file heading, if appl
+- *csharp* contains C# language samples.
 
-**Note:** Currently DocFX requires the .NET Framework on Windows or Mono (for Linux or macOS). We hope to port it to .NET Core in the future. 
+   * The *csharp* directory contains samples where the purpose is to explain the C# language. While these samples will use frameworks and libraries, their focus is on the C# language. Our CI build server builds these samples on multiple supported platforms. Therefore, every sample must be configured to build on Linux, Mac, and Windows.
 
-You can build and preview the resulting site locally using a built-in web server. Navigate to the core-docs folder on your machine and type the following command:
+- *framework* contains .NET Framework samples. These are referenced by topics under many different locations in the documentation. These samples build only on Windows.
 
-```
-docfx -t default --serve
-```
-	
-This starts the local preview on [localhost:8080](http://localhost:8080). You can then view the changes by going to `http://localhost:8080/[path]`, such as http://localhost:8080/articles/welcome.html.
+   * The *framework* directory contains samples that highlight .NET with platform dependencies. These may include migration samples, platform specific samples, or other samples that require the framework or the Windows-based tools. Our CI build server builds these samples on Windows only. Do not place any samples here that should be checked for cross-platform builds.
 
-**Note:** the local preview currently doesn't contain any themes at the moment so the look and feel won't be the same as in the documentation site. We're working towards fixing that experience.
+- *snippets* contains the code snippets used throughout the documentation. Inside the *snippets* folder, you'll see folder names that identify the language of the code snippet.
 
-## Contributing to samples
+We will expand these directories as the docs repository adds new content. For example, we will add Xamarin directories, like `xamarin-ios` and `xamarin-android` directories. 
 
-We encourage samples for all the topics in .NET documentation. Readers can download and run the samples. They
-can explore the code and experiment with their own changes.
+Sample code may fit more than one of these areas. In those cases, place the sample so it matches
+the topics you are covering in your documents. Ask yourself what readers will learn from reading
+your topic. What will they learn from building and running your sample?
 
-All samples should be complete applications or libraries. Where the sample creates a library,
-it should include unit tests or an application that lets readers run the code.
+Each complete sample that you create should contain a `readme.md` file. This file should
+contain a short description of the sample (one or two paragraphs). Your readme.md
+should tell readers what they will learn by exploring this sample. The readme.md file should also contain
```

Comment:
actually, we standardized on using italic for file names to use the same convention ASP.NET had been using for a long time. I'll change and merge.